usaspending_api/agency/v2/views/budgetary_resources.py

```diff
@@ -66,6 +85,7 @@ def get_agency_budgetary_resources(self):
                 "agency_budgetary_resources": x["agency_budgetary_resources"],
                 "agency_total_obligated": x["agency_total_obligated"],
                 "total_budgetary_resources": resources.get(x["submission__reporting_fiscal_year"]),
+                "agency_obligation_by_period": self.get_periods_by_year(x["submission__reporting_fiscal_year"]),
```

Comment:
This line is resulting in a DB query to get fired off for each FY in addition to the query at the beginning of this method. This endpoint in QAT right now completes in about .5 seconds. My test against QAT had this going up to 2.5 seconds. That isn't terrible, but is a noticeable jump. Is it possible to change this to minimize queries against the DB?

Reply:
I've pushed some changes that make the query run roughly the same when testing against the QAT db locally (3.45 s v. 3.54 s), but please double check since those are both worse times than you saw while testing